index.js

```diff
@@ -30,13 +30,26 @@ function injectAsync(source) {
             switch (opts.type) {
                 case undefined:
                 case 'tile':
-                    source.getTile(opts.z, x, y, (err, data, hdrs) => {
+                    var cb = function(err, data, hdrs) {
                         if (err) {
                             reject(err);
                         } else {
                             resolve({data: data, headers: hdrs});
                         }
-                    });
+                    };
+
+                    // fallback for tilelive-vector
```

Comment:
What is this fallback about?  `cb` gets used as either an opts structure, or a callback?

Reply:
tilelive-vector expects additional parameters to be added as structure to the callback because the default tilelive interface looks like `getTile = function(z, x, y, callback)` the kartotherian stack replaces this default interface with the `getAsync(opts)` interface where everything is passed as opts structure.

So to make the legacy tilelive-vector compatible with that approach, the relevant opts structure properties need to be forwarded as properties on the callback function.

Reply:
See also the longer discussion on https://github.com/mapbox/tilelive/issues/189

Reply:
Thanks.  It looks backwards-compatible anyway, which is good enough for now...